blockchain/blockchain.go

```diff
@@ -622,6 +651,21 @@ func stateUpdateByHash(txn db.Transaction, hash *felt.Felt) (*core.StateUpdate,
 	})
 }
 
+func l1MsgHashes(txn db.Transaction, l1HandlerMsgHash *common.Hash) (*felt.Felt, error) {
+	var raw []byte
+	err := txn.Get(db.L1HandlerTxnHashByMsgHash.Key(l1HandlerMsgHash.Bytes()), func(val []byte) error {
+		if len(val) == 0 {
```

Comment:
In principle it shouldn't (I don't think we should ever store a key-value pair where the value is empty, just wanted to be cautious).

Reply:
Since you are in control over storing the hash isn't it safe you assume that you will not store an empty hash?

Such a check should exit in the store function, which then leads you to assume that there will be no key which has an empty value.

Also, `db.ErrKeyNotFound` is an incorrect error to return since the DB found the key but there was no value. Outside of the db package `ErrKeyNotFound` should only be used to check an error not return an error.

Also, it would be worth checking if the DB creates an entry if the value is nil. I believe some tests are dealing with this in the DB package.

IMO this `if` statement should be removed as it is superfluous code.

Reply:
Actually, if the key-value doesn't exist `val` will have length 0, in which case we should return `ErrKeyNotFound` imo.

Reply:
Yup, you're right! I've committed your changes